src/core/request.js

```diff
@@ -79,17 +79,21 @@ Request.prototype.request = function(servers, callback_) {
     // just in case
     this.emit = _.noop;
     this.cancel();
+    this.remote.removeListener('connected', doRequest);
   }, this._timeout);
 
-  function onResponse() {
-    clearTimeout(timeout);
-  }
-
   if (this.remote.isConnected()) {
     this.remote.on('connected', doRequest);
   }
 
-  this.once('response', onResponse);
+  function onRemoteError() {}
+  this.remote.once('error', onRemoteError); // e.g. rate-limiting slowDown error
+
+  this.once('response', () => {
+    clearTimeout(timeout);
```

Comment:
They are not processed by `Request`. `Remote` has `switch` that routes special all other types of messages other than `response` to special handler [here](https://github.com/ripple/ripple-lib/blob/develop/src/core/remote.js#L499)
And `Server`, after receiving `response` and routing it to `Request`, [deletes](https://github.com/ripple/ripple-lib/blob/develop/src/core/server.js#L661) that `Request` object from own list, so if server will send another `response` message with the same id, `Server` will throw it as unexpected.